pkg/network/tracer/tracer_windows.go

```diff
@@ -52,13 +57,85 @@ type Tracer struct {
 	destExcludes   []*network.ConnectionFilter
 }
 
+func isClosedSourceAllowed() bool {
+	ddRegKey := `SOFTWARE\DataDog\Datadog Agent`
+	regKey, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\DataDog\Datadog Agent`, registry.QUERY_VALUE)
+	if err != nil {
+		log.Warnf("unable to open registry key %s: %v", ddRegKey, err)
+		return false
+	}
+	defer regKey.Close()
+
+	ddRegName := "AllowClosedSource"
+	val, _, err := regKey.GetIntegerValue("AllowClosedSource")
+	if err != nil {
+		log.Warnf("unable to get value for %s: %v", ddRegName, err)
+		return false
+	}
+
+	if val == 1 {
+		log.Infof("%s is set; closed-source software allowed", ddRegName)
+		return true
+	} else if val == 0 {
+		log.Infof("%s is not set; closed-source software not allowed", ddRegName)
+		return false
+	} else {
+		log.Infof("Unexpected value set for %s: %d", ddRegName, ddRegName)
+		return false
+	}
+}
+
 // NewTracer returns an initialized tracer struct
 func NewTracer(config *config.Config) (*Tracer, error) {
 	di, err := network.NewDriverInterface(config, driver.NewHandle)
 
 	if err != nil && errors.Is(err, syscall.ERROR_FILE_NOT_FOUND) {
 		log.Debugf("could not create driver interface: %v", err)
-		return nil, fmt.Errorf("The Windows driver was not installed, reinstall the Datadog Agent with network performance monitoring enabled")
+		// check registry to see if closed source is allowed
+		if isClosedSourceAllowed() {
+			manager, err := winutil.OpenSCManager(windows.SC_MANAGER_CONNECT)
+			if err != nil {
+				return nil, err
+			}
+			defer manager.Disconnect()
+
+			ddnpmServiceAccess := windows.SERVICE_CHANGE_CONFIG | windows.SERVICE_START
+			ddnpmServiceHandle, err := winutil.OpenService(manager, "ddnpm", uint32(ddnpmServiceAccess))
+			if err != nil {
+				return nil, err
+			}
+			defer ddnpmServiceHandle.Close()
+
+			newConfig := mgr.Config{
+				StartType: mgr.StartManual,
+			}
+			err = ddnpmServiceHandle.UpdateConfig(newConfig)
+			if err != nil {
+				log.Warnf("could not enable %s: %v", err)
+				return nil, err
+			}
+			err = ddnpmServiceHandle.Start()
+			if err != nil {
+				log.Warnf("Unable to start ddnpm driver")
+				return nil, err
+			}
+
+			sysProbeServiceAccess := windows.SERVICE_CHANGE_CONFIG
+			sysProbeServiceHandle, err := winutil.OpenService(manager, "datadog-systemprobe", uint32(sysProbeServiceAccess))
+			if err != nil {
+				return nil, err
+			}
+			defer sysProbeServiceHandle.Close()
+			newConfig = mgr.Config{
+				Dependencies: []string{"ddnpm"},
```

Comment:
this will wipe out the other dependency (on `datadog-agent`), or any other existing dependency.  Should get the current dependencies, make sure ddnpm isn't in there, and then add it.

Reply:
At the time, I wasn't sure if `UpdateConfig` would only apply the updates to the specified elements (and leave the others alone). I looked after the push and it certainly will wipe them 🙃. For the other entries, I'll need to use `windows.SERVICE_NO_CHANGE`.